From 5199207edfaf7289eb350708b6c2e719a16bf526 Mon Sep 17 00:00:00 2001
From: "Kothapeta, BikshapathiX" <bikshapathix.kothapeta@intel.com>
Date: Fri, 20 Nov 2020 11:12:30 +0530
Subject: [PATCH 4/5] Remove rule for dumpstate_dropbox and logsvc

Remove bellow rule for logsvc to fix cts test errors:
allow logsvc self:capability { sys_nice };

Remove bellow rule for dumpstate_dropbox to fix cts test errors:
allow dumpstate_dropbox self:capability { chown }

update vendor_init to dontaudit and remove unused typeattribute.

Tracked-On: https://jira01.devtools.intel.com/browse/OAM-71987
Signed-off-by: Xinanx, Luo xinanx.luo@intel.com
Signed-off-by: Kothapeta, BikshapathiX <bikshapathix.kothapeta@intel.com>

diff --git a/coredump/vendor_init.te b/coredump/vendor_init.te
index 0919fe1..6f7d811 100644
--- a/coredump/vendor_init.te
+++ b/coredump/vendor_init.te
@@ -1,5 +1,5 @@
 userdebug_or_eng(`
-  allow vendor_init coredump_log_file:dir create_dir_perms;
+  dontaudit vendor_init coredump_log_file:dir create_dir_perms;

   module_only(`debug_coredump', `
     permissive vendor_init;
diff --git a/crashlogd/dumpstate_dropbox.te b/crashlogd/dumpstate_dropbox.te
index e6f9d5b..8fb1a2d 100644
--- a/crashlogd/dumpstate_dropbox.te
+++ b/crashlogd/dumpstate_dropbox.te
@@ -6,7 +6,7 @@ userdebug_or_eng(`

   permissive dumpstate_dropbox;

-  allow dumpstate_dropbox self:capability { chown };
+  dontaudit dumpstate_dropbox self:capability { chown };

   allow domain dumpstate_dropbox:fd use;

@@ -33,9 +33,6 @@ userdebug_or_eng(`
   dontaudit dumpstate_dropbox binder_device: chr_file *;
   dontaudit dumpstate_dropbox default_prop:file *;
   dontaudit dumpstate_dropbox {fs_type dev_type file_type}:dir_file_class_set *;
-
-  module_only(`debug_crashlogd', `
-    allow dumpstate_dropbox log_file:dir ra_dir_perms;
-    allow dumpstate_dropbox log_file:file create_file_perms;
-  ')
+  dontaudit dumpstate_dropbox log_file:dir ra_dir_perms;
+  dontaudit dumpstate_dropbox log_file:file create_file_perms;
 ')
diff --git a/crashlogd/logsvc.te b/crashlogd/logsvc.te
index 3ba0cf4..43ded0a 100644
--- a/crashlogd/logsvc.te
+++ b/crashlogd/logsvc.te
@@ -4,7 +4,7 @@ type logsvc_exec, exec_type, file_type, vendor_file_type;
 init_daemon_domain(logsvc);

 userdebug_or_eng(`
-  allow logsvc self:capability { sys_nice };
+  dontaudit logsvc self:capability{ sys_nice };
   allow logsvc self:capability2 syslog;

   allow logsvc logdr_socket:sock_file write;
diff --git a/crashlogd/vendor_init.te b/crashlogd/vendor_init.te
index 59e0a53..4e75a11 100644
--- a/crashlogd/vendor_init.te
+++ b/crashlogd/vendor_init.te
@@ -1,17 +1,17 @@
 set_prop(vendor_init, vendor_core_prop)
 set_prop(vendor_init, vendor_crashlogd_prop)

-allow vendor_init log_file:dir create_dir_perms;
+dontaudit vendor_init log_file:dir create_dir_perms;
 set_prop(vendor_init, vendor_apklogfs_prop)
 set_prop(vendor_init, vendor_aplogfs_prop)
 set_prop(vendor_init, logpersistd_logging_prop)

 userdebug_or_eng(`
-  allow vendor_init cache_file:lnk_file r_file_perms;
-  allow vendor_init cache_file:dir create_dir_perms;
-  allow vendor_init log_file:dir { relabelfrom relabelto };
-  allow vendor_init log_file:lnk_file create_file_perms;
-  allow vendor_init cache_file:dir { relabelfrom relabelto };
+  dontaudit vendor_init cache_file:lnk_file r_file_perms;
+  dontaudit vendor_init cache_file:dir create_dir_perms;
+  dontaudit vendor_init log_file:dir { relabelfrom relabelto };
+  dontaudit vendor_init log_file:lnk_file create_file_perms;
+  dontaudit vendor_init cache_file:dir { relabelfrom relabelto };
 # data_between_core_and_vendor_violators
-  allow vendor_init system_data_file:dir create_dir_perms;
+  dontaudit vendor_init system_data_file:dir create_dir_perms;
 ')
diff --git a/crashlogd/violators_blacklist.te b/crashlogd/violators_blacklist.te
deleted file mode 100644
index 2f56a4a..0000000
--- a/crashlogd/violators_blacklist.te
+++ /dev/null
@@ -1,7 +0,0 @@
-userdebug_or_eng(`
-
-  typeattribute dumpstate_dropbox data_between_core_and_vendor_violators;
-
-')
-
-typeattribute logsvc data_between_core_and_vendor_violators;
diff --git a/debug-phonedoctor/violators_blacklist.te b/debug-phonedoctor/violators_blacklist.te
deleted file mode 100644
index 86e52c1..0000000
--- a/debug-phonedoctor/violators_blacklist.te
+++ /dev/null
@@ -1,2 +0,0 @@
-typeattribute installd data_between_core_and_vendor_violators;
-
diff --git a/graphics/mesa/vendor_init.te b/graphics/mesa/vendor_init.te
index 0ea3b84..cc5eba7 100644
--- a/graphics/mesa/vendor_init.te
+++ b/graphics/mesa/vendor_init.te
@@ -1,6 +1,6 @@
-allow vendor_init self:capability { sys_module net_raw };
-allow vendor_init vendor_file:system module_load;
-allow vendor_init debugfs_tracing_instances:dir write;
-allow vendor_init mediaserver:process setsched;
-allow vendor_init self:udp_socket create;
-allow vendor_init coreu_data_file:dir create_dir_perms;
+dontaudit vendor_init self:capability { sys_module net_raw };
+dontaudit vendor_init vendor_file:system module_load;
+dontaudit vendor_init debugfs_tracing_instances:dir write;
+dontaudit vendor_init mediaserver:process setsched;
+dontaudit vendor_init self:udp_socket create;
+dontaudit vendor_init coreu_data_file:dir create_dir_perms;
diff --git a/graphics/mesa_acrn/violators_blacklist.te b/graphics/mesa_acrn/violators_blacklist.te
index 20a504e..084022e 100644
--- a/graphics/mesa_acrn/violators_blacklist.te
+++ b/graphics/mesa_acrn/violators_blacklist.te
@@ -1,2 +1 @@
 typeattribute hal_graphics_composer_default data_between_core_and_vendor_violators;
-typeattribute vendor_init data_between_core_and_vendor_violators;
diff --git a/graphics/software/violators_blacklist.te b/graphics/software/violators_blacklist.te
index 20a504e..084022e 100644
--- a/graphics/software/violators_blacklist.te
+++ b/graphics/software/violators_blacklist.te
@@ -1,2 +1 @@
 typeattribute hal_graphics_composer_default data_between_core_and_vendor_violators;
-typeattribute vendor_init data_between_core_and_vendor_violators;
diff --git a/hdcpd/vendor_init.te b/hdcpd/vendor_init.te
index 6577d73..88bbc05 100644
--- a/hdcpd/vendor_init.te
+++ b/hdcpd/vendor_init.te
@@ -1 +1 @@
-allow vendor_init hdcpd_data_file:dir create_dir_perms;
+dontaudit vendor_init hdcpd_data_file:dir create_dir_perms;
diff --git a/hdcpd/violators_blacklist.te b/hdcpd/violators_blacklist.te
deleted file mode 100644
index 877fced..0000000
--- a/hdcpd/violators_blacklist.te
+++ /dev/null
@@ -1 +0,0 @@
-typeattribute vendor_init data_between_core_and_vendor_violators;
diff --git a/pstore/vendor_init.te b/pstore/vendor_init.te
index 0a6bb14..2ea07fe 100644
--- a/pstore/vendor_init.te
+++ b/pstore/vendor_init.te
@@ -1,3 +1,3 @@
 userdebug_or_eng(`
-  allow vendor_init pstore-clean_data_file:dir create_dir_perms;
+  dontaudit vendor_init pstore-clean_data_file:dir create_dir_perms;
 ')
diff --git a/pstore/violators_blacklist.te b/pstore/violators_blacklist.te
deleted file mode 100644
index e3f4065..0000000
--- a/pstore/violators_blacklist.te
+++ /dev/null
@@ -1,2 +0,0 @@
-typeattribute pstore-clean data_between_core_and_vendor_violators;
-
--
2.29.2

